spec/models/user_spec.rb

```diff
@@ -519,12 +519,39 @@
     end
   end
 
+  describe '#gift_sticker' do
+    before do
+      allow(UserStateTransitionSegmentService)
+        .to receive(:gift_sticker).and_return(true)
+    end
+
+    context 'the user is in the incompleted state' do
+      let(:user) { FactoryBot.create(:user, :incompleted) }
+
+      context 'there are sticker coupons available' do
```

Comment:
Add a context for `there are shirt coupons available`

Reply:
Should this case be added when we add a `gifted_shirt` state

Reply:
it should be added now to ensure the behavior is correct. The test might change when you introduce a `gifted_shirt` state.